program/src/state.rs

```diff
@@ -96,7 +96,7 @@ pub struct EncryptedKeyData {
     /// The identifier of the key that this key is encrypted with
     pub kid: Kid,   // TODO needed?
```

Comment:
What's the type here and should that be also converted to a fixed-size byte representation?